beprof/curve.py

```diff
@@ -51,8 +53,10 @@ def __new__(cls, input_array, **meta):
         return obj
 
     def __array_finalize__(self, obj):
-        if obj is None:
+        # print("Here I am in Curve.__array_finalize__ obj: ", type(obj))
+        if obj is None: # what generally means the object was created using explicit constructor
             return
+        self.metadata = getattr(obj, 'metadata', None)
```

Comment:
There is an inconsistency when you are dealing with lack of metadata:

```
>>> import numpy as np
>>> from beprof import profile
>>> from beprof import curve
>>> n = np.asarray( [[0,0], [2,2]])
>>> nc = n.view(curve.Curve)
>>> print(n)
[[0 0]
 [2 2]]
>>> print(nc)
shape: (2, 2)
X : [0.000,2.000]
Y : [0.000000,2.000000]
Metadata : None
>>> nc.metadata
>>> c = curve.Curve( [[0,0], [1,1]])
>>> print(c)
shape: (2, 2)
X : [0.000,1.000]
Y : [0.000000,1.000000]
Metadata : {}
>>> c.metadata
{}
```

Shouldn't you rather take empty directory as default value, rather than none ?

Check also `Profile`